### PR TITLE
Make `ApplicationResponder` public

### PR DESCRIPTION
**Pull request by MrLotU**
I'm working on a monitoring module for Vapor over at https://github.com/MrLotU/VaporMonitoring (still a WIP) and for it to properly function I need access to `ApplicationResponder`, thus this PR

### Checklist

- [x] Circle CI is passing (code compiles and passes tests).
- [x] There are no breaking changes to public API.
- [x] New test cases have been added where appropriate.
- [x] All new code has been commented with doc blocks `///`.

### CODE REVIEW

**Comment by tanner0101** [comment]
Hi, thanks for the PR. Could you explain the use case behind this? I like to use `public` as sparingly as possible.

**Comment by MrLotU** [comment]
Sure thing, I'm working on integrating [SwiftMetrics](https://github.com/RuntimeTools/SwiftMetrics) with Vapor for a monitoring/metrics dashboard + Prometheus support. Current progress can be seen [here](https://github.com/MrLotU/VaporMonitoring) (name might change). The reason I made `ApplicationResponder` public is so that I can use it to, in the background, power my `MonitoredResponder`. I do the same with `MonitoredRouter` where I use `EngineRouter` as actual router and just add the monitoring.

If there's any other way to achieve this without having to declare `ApplicationResponder` public, I'm all ears, but I couldn't think of any.

**Comment by tanner0101** [comment]
You should be able to call just `make(Responder.self)` to get a generic `Responder` (you shouldn't need specifically an `ApplicationResponder` unless I'm missing something).

**Comment by tanner0101** [comment]
Oops, hit wrong button.

**Comment by MrLotU** [comment]
That's true, although I wasn't sure if making a `Responder.self` would do, since I'm registering `MonitoredResponder.self` as `Responder.self` and wasn't sure if it could've conflicted if for some reason someone made a `Responder.self` twice (and if, in that case, `MonitoredResponder` would be passed through as internal to `MonitoredResponder`)

**Comment by MrLotU** [comment]
@tanner0101 Any news on this? It's currently breaking VaporMonitoring

File: Sources/Vapor/Response/ApplicationResponder.swift [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
+        return ApplicationResponder(router, middleware: middleware)
     }
 
     /// Wrapped `Responder`.
     private let responder: Responder
 
     /// Creates a new `ApplicationResponder`.
-    init(_ responder: Responder) {
-        self.responder = responder
+    public init(_ router: Router, middleware: [Middleware]) {
```
Comment (tanner0101) [inline Sources/Vapor/Response/ApplicationResponder.swift]: Either both should have labels or both not I think. Also middleware can have a default arg to `[]`.

**Comment by penny-coin** [comment]
Hey @MrLotU, you just merged a pull request, have a coin! 

You now have 14 coins.

**Comment by MrLotU** [comment]
Thanks!

**Pull request merged**